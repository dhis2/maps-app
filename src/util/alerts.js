// Returns array of all alerts from map config and layers (mapViews)
export const getMapAlerts = mapConfig =>
    []
        .concat(...(mapConfig && mapConfig.alerts ? mapConfig.alerts : []))
        .concat(
            ...(mapConfig &&
                mapConfig.mapViews &&
                mapConfig.mapViews
                    .filter(layer => layer.alerts)
                    .map(layer => layer.alerts))
        );
