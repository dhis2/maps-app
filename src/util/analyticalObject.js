import { THEMATIC_LAYER } from '../constants/layers.js'
import { getPeriodNameFromId, getDimensionsFromFilters } from './analytics.js'
import { cleanDimension } from './favorites.js'
import { loadDataItemLegendSet } from './legend.js'

export const USER_DATASTORE_NAMESPACE = 'analytics'
export const CURRENT_AO_KEY = 'currentAnalyticalObject'

export const APP_URLS = {
    CHART: 'dhis-web-data-visualizer',
    PIVOT: 'dhis-web-pivot',
}

// Combines all dimensions in columns, rows and filters
const getDimensionsFromAnalyticalObject = (ao = {}) => {
    const { columns = [], rows = [], filters = [] } = ao
    return [...columns, ...rows, ...filters]
}

// Returns the data items of the first dx dimension in an analytical object
export const getDataDimensionsFromAnalyticalObject = (ao) => {
    const dims = getDimensionsFromAnalyticalObject(ao)
    const dataDim = dims.find((i) => i.dimension === 'dx')

    // We only use the first dx dimension
    return dataDim ? dataDim.items : []
}

// Returns a thematic layer config from an analytical object
export const getThematicLayerFromAnalyticalObject = async (
    ao = {},
    dataId,
    isVisible = true
) => {
    const { yearlySeries, aggregationType = 'DEFAULT' } = ao
    const dataDims = getDataDimensionsFromAnalyticalObject(ao)
    const dims = getDimensionsFromAnalyticalObject(ao)
    const orgUnits = dims.find((i) => i.dimension === 'ou')
    const filters = getDimensionsFromFilters(dims) // Dynamic dimension filters
    let period = dims.find((i) => i.dimension === 'pe')
    let dataDim = dataDims[0]

    if (dataId) {
        dataDim = dataDims.find((item) => item.id === dataId)
    }

    if (!dataDim || !orgUnits || !period) {
        return
    }

    // Load default legend set for selected data dimension
    const legendSet = await loadDataItemLegendSet(dataDim)

    // Currently we only support one period in map filters so we select the first
    if (yearlySeries && yearlySeries.length) {
        period = {
            dimension: 'pe',
            items: [
                {
                    id: yearlySeries[0],
                    name: getPeriodNameFromId(yearlySeries[0]),
                },
            ],
        }
    }

    return {
        layer: THEMATIC_LAYER,
        columns: [{ dimension: 'dx', items: [dataDim] }],
        rows: [orgUnits],
        filters: [period, ...filters],
        aggregationType,
        legendSet,
        isVisible,
        opacity: 0.9,
    }
}

// Translates a thematic layer to an analytical object
export const getAnalyticalObjectFromThematicLayer = (layer = {}) => {
    const {
        columns = [],
        rows = [],
        filters = [],
        aggregationType = 'DEFAULT',
    } = layer

    return {
        columns,
        rows: rows.map(cleanDimension),
        filters,
        aggregationType,
    }
}
