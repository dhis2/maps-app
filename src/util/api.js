import isString from 'lodash/fp/isString';
import isObject from 'lodash/fp/isObject';
import { config } from 'd2/lib/d2';

// TODO: Channel all api request through d2

export const apiFetch = async (url, method, body) => {
    const options = {
        headers: {},
    };

    if (config.context && config.context.auth) {
        options.headers['Authorization'] = 'Basic ' + btoa(config.context.auth);
    } else {
        options.credentials = 'include';
    }

    if (method && body) {
        options.method = method;

        if (isString(body)) {
            options.headers['Content-Type'] = 'text/html';
            options.body = body;
        } else if (isObject(body)) {
            options.headers['Content-Type'] = 'application/json';
            options.body = JSON.stringify(body);
        }
    }

    return fetch(encodeURI(config.baseUrl + url), options)
        .then(
            response =>
                method !== 'PUT' && method !== 'PATCH' && method !== 'DELETE'
                    ? response.json()
                    : response
        ) // Avoids error
        .catch(error => console.log('Error: ', error)); // TODO: Better error handling
};
