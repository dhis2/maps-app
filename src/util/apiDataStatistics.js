export const dataStatisticsMutation = {
    resource: 'dataStatistics',
    params: ({ id }) => ({
        favorite: id,
        eventType: 'MAP_VIEW',
    }),
    type: 'create',
}
