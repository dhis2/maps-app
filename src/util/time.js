import i18n from '@dhis2/d2-i18n';

const DEFAULT_LOCALE = 'en';

/**
 * Converts a date string or timestamp to a date object
 * @param {String|Number|Date} date
 * @returns {String}
 */
export const toDate = date => new Date(date);

// Simple check if the date part is correctly formatted
const shortDateRegexp = /^\d{4}-\d{2}-\d{2}$/;

/**
 * Checks if the date format is valid
 * @param {String} dateString
 * @returns {String}
 */
export const isValidDateFormat = dateString =>
    shortDateRegexp.test(dateString.substr(0, 10));

/**
 * Formats a date string or timestamp into format used by DHIS2 and <input> date
 * @param {String|Number|Date} date
 * @returns {String}
 */
export const formatDate = date => {
    const dateObj = toDate(date);
    const year = dateObj.getFullYear();
    const month = ('0' + (dateObj.getMonth() + 1)).slice(-2);
    const day = ('0' + dateObj.getDate()).slice(-2);
    return `${year}-${month}-${day}`; // xxxx-xx-xx
};

/**
 * Simple fallback date format if Intl is not supported
 * @param {String} dateString
 * @returns {String}
 */
const fallbackDateFormat = dateString => dateString.substr(0, 10);

/**
 * Returns true if the Internationalization API is supported
 * @returns {Boolean}
 */
export const hasIntlSupport =
    typeof global.Intl !== 'undefined' && Intl.DateTimeFormat;

/**
 * Formats a date string or timestamp to the default display format: 13 Aug 2018 (en locale)
 * @param {String} dateString
 * @param {String} locale
 * @returns {String}
 */
export const formatLocaleDate = (dateString, locale) =>
    hasIntlSupport
        ? new Intl.DateTimeFormat(locale || i18n.language || DEFAULT_LOCALE, {
              year: 'numeric',
              month: 'short',
              day: 'numeric',
          }).format(toDate(dateString))
        : fallbackDateFormat(dateString);

/**
 * Formats a date range
 * @param {String|Number} startDate
 * @param {String|Number} endDate
 * @param {String} locale
 * @returns {String}
 */
export const formatStartEndDate = (startDate, endDate, locale) => {
    const loc = locale || i18n.language || DEFAULT_LOCALE;
    return `${formatLocaleDate(startDate, loc)} - ${formatLocaleDate(
        endDate,
        locale
    )}`;
};

/**
 * Checks for errors for start and end date strings or timestamps
 * @param {String} startDate
 * @param {String} endDate
 * @returns {String|null}
 */
export const getStartEndDateError = (startDate, endDate) => {
    if (!isValidDateFormat(startDate)) {
        return i18n.t('Start date is invalid');
    } else if (!isValidDateFormat(endDate)) {
        return i18n.t('End date is invalid');
    } else if (toDate(endDate) < toDate(startDate)) {
        return i18n.t('End date cannot be earlier than start date');
    }
    return null;
};

/**
 * Returns the year of the date, or the current year of no date is passed
 * @param {String|Number|Date} startDate
 * @returns {Number}
 */
export const getYear = date => toDate(date || new Date()).getFullYear();
