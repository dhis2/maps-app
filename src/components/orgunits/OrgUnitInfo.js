import React from 'react';
import PropTypes from 'prop-types';
import i18n from '@dhis2/d2-i18n';
import { config } from 'd2';
import { IconDimensionOrgUnit16 } from '@dhis2/ui';
import ListItem from '../core/ListItem';
import { formatDate } from '../../util/time';
import { numberPrecision } from '../../util/numbers';
import styles from './styles/OrgUnitInfo.module.css';

export const coordFormat = numberPrecision(6); // Meter precision for longitude an latitude

/*
 * Displays the fixed information for an org unit together with org unit groups (above) and metadata attributes (below)
 *
 * Available org unit info fields:
 * https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/orgunitprofile/OrgUnitInfo.java
 */
const OrgUnitInfo = ({
    address,
    attributes,
    closedDate,
    code,
    comment,
    contactPerson,
    description,
    email,
    groupSets,
    id,
    imageId,
    latitude,
    level,
    levelName,
    longitude,
    name,
    openingDate,
    parentName,
    phoneNumber,
    shortName,
    url,
}) => (
    <div className={styles.info}>
        {imageId && (
            <img
                src={`${config.baseUrl}/fileResources/${imageId}/data`}
                alt={i18n.t('Image of the organisation unit')}
            />
        )}
        {name && <h3>{name}</h3>}
        {(level || levelName) && (
            <div className={styles.level}>
                <IconDimensionOrgUnit16 />
                {i18n.t('Level')}:{' '}
                {levelName
                    ? `${levelName}${level ? ` (${level})` : ''}`
                    : level}
            </div>
        )}
        {parentName && (
            <div className={styles.level}>
                <IconDimensionOrgUnit16 />
                {i18n.t('Parent')}: {parentName}
            </div>
        )}
        {description && <div className={styles.desc}>{description}</div>}
        <div className={styles.list}>
            {groupSets.map(({ id, label, value }) => (
                <ListItem key={id} label={label}>
                    {value}
                </ListItem>
            ))}
            <ListItem label={i18n.t('Code')}>{code}</ListItem>
            <ListItem label={i18n.t('Short name')}>{shortName}</ListItem>
            <ListItem label={i18n.t('Opening date')} formatter={formatDate}>
                {openingDate}
            </ListItem>
            <ListItem label={i18n.t('Closed date')} formatter={formatDate}>
                {closedDate}
            </ListItem>
            {url && (
                <ListItem label={i18n.t('URL')}>
                    <a href={url} target="_blank" rel="noreferrer">
                        {i18n.t('Link')}
                    </a>
                </ListItem>
            )}
            <ListItem label={i18n.t('Contact')}>{contactPerson}</ListItem>
            <ListItem label={i18n.t('Email')}>{email}</ListItem>
            <ListItem label={i18n.t('Address')}>{address}</ListItem>
            <ListItem label={i18n.t('Phone')}>{phoneNumber}</ListItem>
            <ListItem label={i18n.t('Longitude')} formatter={coordFormat}>
                {longitude}
            </ListItem>
            <ListItem label={i18n.t('Latitude')} formatter={coordFormat}>
                {latitude}
            </ListItem>
            <ListItem label={i18n.t('ID')}>{id}</ListItem>
            <ListItem label={i18n.t('Comment')}>{comment}</ListItem>
            {attributes.map(({ id, label, value }) => (
                <ListItem key={id} label={label}>
                    {value}
                </ListItem>
            ))}
        </div>
    </div>
);

OrgUnitInfo.propTypes = {
    address: PropTypes.string,
    attributes: PropTypes.array.isRequired,
    closedDate: PropTypes.string,
    code: PropTypes.string,
    comment: PropTypes.string,
    contactPerson: PropTypes.string,
    description: PropTypes.string,
    email: PropTypes.string,
    groupSets: PropTypes.array.isRequired,
    id: PropTypes.string.isRequired,
    imageId: PropTypes.string,
    latitude: PropTypes.number,
    level: PropTypes.number,
    levelName: PropTypes.string,
    longitude: PropTypes.number,
    name: PropTypes.string,
    openingDate: PropTypes.string,
    parentName: PropTypes.string,
    phoneNumber: PropTypes.string,
    shortName: PropTypes.string,
    url: PropTypes.string,
};

export default OrgUnitInfo;
