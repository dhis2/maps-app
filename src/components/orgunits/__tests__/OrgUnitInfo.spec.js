import React from 'react';
import { shallow } from 'enzyme';
import OrgUnitInfo, { coordFormat } from '../OrgUnitInfo';
import { formatDate } from '../../../util/time';

const defaultProps = {
    id: 'DiszpKrYNg8',
    groupSets: [],
    attributes: [],
};

const groupSets = [
    {
        id: 'Bpx0589u8y0',
        label: 'Facility Ownership',
        value: 'Public facilities',
    },
    {
        id: 'J5jldMd8OHv',
        label: 'Facility Type',
        value: 'CHC',
    },
];

const attributes = [
    {
        id: 'n2xYlNbsfko',
        label: 'NGO ID',
        value: 'ddASDd2312',
    },
];

// Helper function to get list item value
const getListItem = (wrapper, label) =>
    wrapper
        .find(`[label="${label}"]`)
        .dive()
        .find('td')
        .prop('children');

describe('Org unit profile (location details)', () => {
    const renderWithProps = props =>
        shallow(<OrgUnitInfo {...defaultProps} {...props} />);

    it('should render org unit name', () => {
        const name = 'Ngelehun CHC';
        const wrapper = renderWithProps({ name });

        expect(wrapper.find('h3').prop('children')).toEqual(name);
    });

    it('should render org unit level', () => {
        const level = 4;
        const levelName = 'Facility';

        expect(
            renderWithProps({ level })
                .find('[className="level"]')
                .prop('children')
        ).toContain(level);

        expect(
            renderWithProps({ levelName })
                .find('[className="level"]')
                .prop('children')
        ).toContain(levelName);

        expect(
            renderWithProps({ level, levelName })
                .find('[className="level"]')
                .prop('children')
        ).toContain(`${levelName} (${level})`);
    });

    it('should render org unit parent name', () => {
        const parentName = 'Badjia';
        const wrapper = renderWithProps({ parentName });

        expect(wrapper.find('[className="level"]').prop('children')).toContain(
            parentName
        );
    });

    it('should render org unit description', () => {
        const description = 'Org unit description';
        const wrapper = renderWithProps({ description });

        expect(wrapper.find('[className="desc"]').prop('children')).toContain(
            description
        );
    });

    it('should render fixed attributes in list', () => {
        const code = 'OU_559';
        const shortName = 'Ngelehun CHC';
        const comment = 'This is a comment';
        const contactPerson = 'John Traore';
        const address = 'Problemveien 7, 0315 Oslo, Norway';
        const email = 'john@dhis2test.org';
        const phoneNumber = '+47 22 85 50 50';
        const wrapper = renderWithProps({
            code,
            shortName,
            comment,
            contactPerson,
            address,
            email,
            phoneNumber,
        });

        expect(getListItem(wrapper, 'Code')).toEqual(code);
        expect(getListItem(wrapper, 'Short name')).toEqual(shortName);
        expect(getListItem(wrapper, 'Comment')).toEqual(comment);
        expect(getListItem(wrapper, 'Contact')).toEqual(contactPerson);
        expect(getListItem(wrapper, 'Address')).toEqual(address);
        expect(getListItem(wrapper, 'Email')).toEqual(email);
        expect(getListItem(wrapper, 'Phone')).toEqual(phoneNumber);
    });

    it('should render formatted dates', () => {
        const openingDate = '1970-01-01T00:00:00.000';
        const closedDate = '2021-06-09T00:00:00.000';
        const wrapper = renderWithProps({ openingDate, closedDate });
        const opening = getListItem(wrapper, 'Opening date');
        const closed = getListItem(wrapper, 'Closed date');

        expect(opening).not.toEqual(openingDate);
        expect(opening).toEqual(formatDate(openingDate));
        expect(closed).not.toEqual(closedDate);
        expect(closed).toEqual(formatDate(closedDate));
    });

    it('should render formatted longitude and latitude', () => {
        const longitude = -11.4197452367;
        const latitude = 8.103932863;
        const wrapper = renderWithProps({ longitude, latitude });
        const lng = getListItem(wrapper, 'Longitude');
        const lat = getListItem(wrapper, 'Latitude');

        expect(lng).not.toEqual(longitude);
        expect(lng).toEqual(coordFormat(longitude));
        expect(lat).not.toEqual(latitude);
        expect(lat).toEqual(coordFormat(latitude));
    });

    it('should render link for org unit url', () => {
        const url = 'https://dhis2.org/';
        const wrapper = renderWithProps({ url });

        expect(
            wrapper
                .find(`[label="URL"]`)
                .dive()
                .find('a')
                .prop('href')
        ).toEqual(url);
    });

    it('should render org unit group memberships', () => {
        const wrapper = renderWithProps({ groupSets });

        groupSets.forEach(({ label, value }) =>
            expect(getListItem(wrapper, label)).toEqual(value)
        );
    });

    it('should render metadata attributes', () => {
        const wrapper = renderWithProps({ attributes });

        attributes.forEach(({ label, value }) =>
            expect(getListItem(wrapper, label)).toEqual(value)
        );
    });
});
