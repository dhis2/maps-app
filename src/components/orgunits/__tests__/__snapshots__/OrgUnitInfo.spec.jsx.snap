// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Org unit profile (location details) renders OrgUnitInfo 1`] = `
<div>
  <div
    class="info"
    data-test="org-unit-info"
  >
    <h3>
      Port Loko
    </h3>
    <div
      class="level"
    >
      <svg
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 1v4H4v2h6V6h4v4h-4V8H4v4h6v-1h4v4h-4v-2H3.5a.5.5 0 01-.492-.41L3 12.5V5H2V1zm7 11h-2v2h2zm0-5h-2v2h2zM5 2H3v2h2z"
          fill="currentColor"
        />
      </svg>
      Level
      :
       
      District (2)
    </div>
    <div
      class="level"
    >
      <svg
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 1v4H4v2h6V6h4v4h-4V8H4v4h6v-1h4v4h-4v-2H3.5a.5.5 0 01-.492-.41L3 12.5V5H2V1zm7 11h-2v2h2zm0-5h-2v2h2zM5 2H3v2h2z"
          fill="currentColor"
        />
      </svg>
      Parent
      : 
      Sierra Leone
    </div>
    <div
      class="desc"
    >
      The district of Port Loko
    </div>
    <div
      class="list"
    >
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Code
            </th>
            <td>
              PL
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Short name
            </th>
            <td>
              PLoko
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              URL
            </th>
            <td>
              <a
                href="portloko.dhis2.org"
                rel="noreferrer"
                target="_blank"
              >
                Link
              </a>
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Contact
            </th>
            <td>
              Johnny
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Email
            </th>
            <td>
              portloko@dhis2.org
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Address
            </th>
            <td>
              Port Loko 1
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Phone
            </th>
            <td>
              123-456-7890
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Longitude
            </th>
            <td>
              -11.419745
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Latitude
            </th>
            <td>
              8.103933
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              ID
            </th>
            <td>
              portlokoid
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Comment
            </th>
            <td>
              This is the comment
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Org unit profile (location details) renders formatted dates if featureType is POINT 1`] = `
<div>
  <div
    class="info"
    data-test="org-unit-info"
  >
    <h3>
      Port Loko
    </h3>
    <div
      class="level"
    >
      <svg
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 1v4H4v2h6V6h4v4h-4V8H4v4h6v-1h4v4h-4v-2H3.5a.5.5 0 01-.492-.41L3 12.5V5H2V1zm7 11h-2v2h2zm0-5h-2v2h2zM5 2H3v2h2z"
          fill="currentColor"
        />
      </svg>
      Level
      :
       
      District (2)
    </div>
    <div
      class="level"
    >
      <svg
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 1v4H4v2h6V6h4v4h-4V8H4v4h6v-1h4v4h-4v-2H3.5a.5.5 0 01-.492-.41L3 12.5V5H2V1zm7 11h-2v2h2zm0-5h-2v2h2zM5 2H3v2h2z"
          fill="currentColor"
        />
      </svg>
      Parent
      : 
      Sierra Leone
    </div>
    <div
      class="desc"
    >
      The district of Port Loko
    </div>
    <div
      class="list"
    >
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Code
            </th>
            <td>
              PL
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Short name
            </th>
            <td>
              PLoko
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Opening date
            </th>
            <td>
              1970-01-01
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Closed date
            </th>
            <td>
              2021-06-09
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              URL
            </th>
            <td>
              <a
                href="portloko.dhis2.org"
                rel="noreferrer"
                target="_blank"
              >
                Link
              </a>
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Contact
            </th>
            <td>
              Johnny
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Email
            </th>
            <td>
              portloko@dhis2.org
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Address
            </th>
            <td>
              Port Loko 1
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Phone
            </th>
            <td>
              123-456-7890
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Longitude
            </th>
            <td>
              -11.419745
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Latitude
            </th>
            <td>
              8.103933
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              ID
            </th>
            <td>
              portlokoid
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Comment
            </th>
            <td>
              This is the comment
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Org unit profile (location details) renders metadata attributes 1`] = `
<div>
  <div
    class="info"
    data-test="org-unit-info"
  >
    <h3>
      Port Loko
    </h3>
    <div
      class="level"
    >
      <svg
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 1v4H4v2h6V6h4v4h-4V8H4v4h6v-1h4v4h-4v-2H3.5a.5.5 0 01-.492-.41L3 12.5V5H2V1zm7 11h-2v2h2zm0-5h-2v2h2zM5 2H3v2h2z"
          fill="currentColor"
        />
      </svg>
      Level
      :
       
      District (2)
    </div>
    <div
      class="level"
    >
      <svg
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 1v4H4v2h6V6h4v4h-4V8H4v4h6v-1h4v4h-4v-2H3.5a.5.5 0 01-.492-.41L3 12.5V5H2V1zm7 11h-2v2h2zm0-5h-2v2h2zM5 2H3v2h2z"
          fill="currentColor"
        />
      </svg>
      Parent
      : 
      Sierra Leone
    </div>
    <div
      class="desc"
    >
      The district of Port Loko
    </div>
    <div
      class="list"
    >
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Code
            </th>
            <td>
              PL
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Short name
            </th>
            <td>
              PLoko
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              URL
            </th>
            <td>
              <a
                href="portloko.dhis2.org"
                rel="noreferrer"
                target="_blank"
              >
                Link
              </a>
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Contact
            </th>
            <td>
              Johnny
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Email
            </th>
            <td>
              portloko@dhis2.org
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Address
            </th>
            <td>
              Port Loko 1
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Phone
            </th>
            <td>
              123-456-7890
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Longitude
            </th>
            <td>
              -11.419745
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Latitude
            </th>
            <td>
              8.103933
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              ID
            </th>
            <td>
              portlokoid
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Comment
            </th>
            <td>
              This is the comment
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              NGO ID
            </th>
            <td>
              ddASDd2312
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Org unit profile (location details) renders org unit group memberships 1`] = `
<div>
  <div
    class="info"
    data-test="org-unit-info"
  >
    <h3>
      Port Loko
    </h3>
    <div
      class="level"
    >
      <svg
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 1v4H4v2h6V6h4v4h-4V8H4v4h6v-1h4v4h-4v-2H3.5a.5.5 0 01-.492-.41L3 12.5V5H2V1zm7 11h-2v2h2zm0-5h-2v2h2zM5 2H3v2h2z"
          fill="currentColor"
        />
      </svg>
      Level
      :
       
      District (2)
    </div>
    <div
      class="level"
    >
      <svg
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 1v4H4v2h6V6h4v4h-4V8H4v4h6v-1h4v4h-4v-2H3.5a.5.5 0 01-.492-.41L3 12.5V5H2V1zm7 11h-2v2h2zm0-5h-2v2h2zM5 2H3v2h2z"
          fill="currentColor"
        />
      </svg>
      Parent
      : 
      Sierra Leone
    </div>
    <div
      class="desc"
    >
      The district of Port Loko
    </div>
    <div
      class="list"
    >
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Facility Ownership
            </th>
            <td>
              Public facilities
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Facility Type
            </th>
            <td>
              CHC
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Code
            </th>
            <td>
              PL
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Short name
            </th>
            <td>
              PLoko
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              URL
            </th>
            <td>
              <a
                href="portloko.dhis2.org"
                rel="noreferrer"
                target="_blank"
              >
                Link
              </a>
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Contact
            </th>
            <td>
              Johnny
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Email
            </th>
            <td>
              portloko@dhis2.org
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Address
            </th>
            <td>
              Port Loko 1
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Phone
            </th>
            <td>
              123-456-7890
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Longitude
            </th>
            <td>
              -11.419745
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Latitude
            </th>
            <td>
              8.103933
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              ID
            </th>
            <td>
              portlokoid
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Comment
            </th>
            <td>
              This is the comment
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Org unit profile (location details) renders with minimal properties 1`] = `
<div>
  <div
    class="info"
    data-test="org-unit-info"
  >
    <div
      class="list"
    >
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              Email
            </th>
            <td>
              johnny@dhis2.org
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="listItem"
      >
        <tbody>
          <tr>
            <th>
              ID
            </th>
            <td>
              portloko123
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
